apps/namadillo/src/App/Governance/ProposalHeader.tsx

```diff
@@ -211,37 +211,63 @@ const StatusLabel: React.FC<{
 };
 
 const Progress: React.FC<{
-  proposal: AtomWithQueryResult<StoredProposal>;
+  proposal: AtomWithQueryResult<Proposal>;
 }> = ({ proposal }) => {
   return (
     <div className="w-full grid grid-cols-2 text-xs">
       <span>Progress</span>
+      <TimeRemaining proposalQuery={proposal} />
       <div className="col-span-2 mt-3 mb-2">
         <ProgressBar cachedProposal={proposal} />
       </div>
       <ProgressStartEnd
         proposal={proposal}
-        epochKey="startEpoch"
+        timeKey="startTime"
         className="col-start-1"
       />
       <ProgressStartEnd
         proposal={proposal}
-        epochKey="endEpoch"
+        timeKey="endTime"
         className="text-right"
       />
     </div>
   );
 };
 
+const TimeRemaining: React.FC<{
+  proposalQuery: AtomWithQueryResult<Proposal>;
+}> = ({ proposalQuery }) => {
+  const text = (() => {
+    if (
+      proposalQuery.status !== "pending" &&
```

Comment:
I think the only possible condition here is the `proposalQuery.status` being equals to `success`

Reply:
Oh nice! That makes more sense.